temporal-sdk/src/main/java/io/temporal/internal/history/VersionMarkerUtils.java

```diff
@@ -75,17 +93,46 @@ public static Integer getVersion(MarkerRecordedEventAttributes markerAttributes)
     return MarkerUtils.getValueFromMarker(markerAttributes, MARKER_VERSION_KEY, Integer.class);
   }
 
+  @Nullable
+  public static boolean getUpsertVersionSA(MarkerRecordedEventAttributes markerAttributes) {
+    Boolean upsertSA =
+        MarkerUtils.getValueFromMarker(markerAttributes, UPSERT_VERSION_SA_KEY, Boolean.class);
+    return upsertSA != null && upsertSA;
+  }
+
   public static RecordMarkerCommandAttributes createMarkerAttributes(
-      String changeId, Integer version) {
+      String changeId, Integer version, Boolean upsertVersionSA) {
     Preconditions.checkNotNull(version, "version");
     Map<String, Payloads> details = new HashMap<>();
     details.put(
         MARKER_CHANGE_ID_KEY, DefaultDataConverter.STANDARD_INSTANCE.toPayloads(changeId).get());
     details.put(
         MARKER_VERSION_KEY, DefaultDataConverter.STANDARD_INSTANCE.toPayloads(version).get());
+    details.put(
+        UPSERT_VERSION_SA_KEY,
+        DefaultDataConverter.STANDARD_INSTANCE.toPayloads(upsertVersionSA).get());
```

Comment:
Can you help me understand the changing of the marker value here. Do other SDKs do this and/or are we using it as a flag of sorts?

Reply:
Yes Go does the same thing, it tells the SDK if it should expect a search attribute after the marker.